### ISSUE TITLE
Individual photos that are part of a stack count against the total count of images for lazy loading

### ISSUE THREAD

**Issue by Morl99** [photoprism/photoprism#510]
At the top of my collection I have quite a few stacked photos, which led to the detection of this bug.  If I switch to the mosaic, only 15 tiles are shown, even though the api call is made with count=60 and offset=0. This is not only confusing, but this breaks the lazy loading, as there is no way to scroll down. See this screenshot for a better understanding (I removed the personal images from the screenshot and replaced it with the blue background, but you can see the stack markers).

![image](https://user-images.githubusercontent.com/475516/94192609-7a21f880-feaf-11ea-8b1b-4f39fd2af7ef.png)

I am running `photoprism/photoprism:20200921`.

I don't get what the frontend is doing here, I see the following 2 API Calls in short succession:
`api/v1/photos?count=360&offset=0&merged=true&country=&camera=0&lens=0&label=&year=0&month=0&color=&order=newest&q=&public=true&quality=3&photo=true`
`api/v1/photos?count=60&offset=60&merged=true&country=&camera=0&lens=0&label=&year=0&month=0&color=&order=newest&q=&public=true&quality=3&photo=true`

The result of the first call actually returns 220 photos, which makes it weird, that the UI only displays 15 items. The second request returns only 4 photos (with a total of 60files included). It looks like, this is a combination of an API bug (the limit should limit the number of photos not files, at least if the ui is set to show stacks as a single item) and a Frontend bug (why are 360 photos requested, but the result seems to be dismissed?)

After scrolling some more, the following sequence of API calls is made:

![image](https://user-images.githubusercontent.com/475516/94196654-fbc85500-feb4-11ea-8fe2-6ef4dbf8a16d.png)

The last api call is taking about 10 seconds, which really impacts performance. Is this a known behavior (e.g. simple implementation vs. good performance), or is here something amiss? It seems really unnessecary for the frontend to fetch all the photo data once we have significantly scrolled, and it looks like some part of the implementation tries to handle the offset correctly. 

Is there something I can do to help?

**Comment by lastzero** [comment]
That's a duplicate of #507. Note that count refers to individual files, not photos / videos which may include multiple files like a RAW file and a JPEG. Our API may also return them unmerged.

**Comment by Morl99** [comment]
I don't feel that this is a duplicate, because #507 specifically states, that this is a problem isolated to the video view, while this issue describes a problem in the main photos view. It might be true, that the root cause is the same, and if this is the case, it is possible to merge the issues, but in that case, I would add the first part of my description as a comment to the issue. But because #507 really does not reveal the root cause, I am not sure if that is really the case.

The second part of this issue is something differnet, and I think it has nothing to do with the first part. I regret mixing those two issues. I also think I understand now, what is going on. The problem I describe is only observable, while I index new photos in the background. This constantly marks the whole collection as dirty, and the frontend reloads the whole image library up to the point of scrolling. So I think this is by design, and I would drop the issue for the time being, because I know it only happens during image ingestion. I still think that moving forward, that is something that should be improved upon, but it needs a redesign of the pagination of the api using cursors instead of offset (see https://developer.twitter.com/en/docs/pagination for example). If you like, we could move this into its own separate issue.

**Comment by lastzero** [comment]
Hopefully I get to this today... I'm confident the root issue is the same.

**Comment by Morl99** [comment]
Alright! If you have a fix I will be happy to test it, just let me know.

**Comment by lastzero** [comment]
Will be deployed for testing with our next Docker image, hopefully later today.

**Comment by lastzero** [comment]
New Docker images available for testing! Note that the password will be reset to its initial state when upgrading. Our Docker images have been optimized for common NAS models. You can now set a specific user and group using the UID and GID environment variables. Feedback welcome.

**Comment by Morl99** [comment]
Thanks for fixing this. I tested it with the same images where I described the issue before, and now it works as expected.
Updating the docker image also was a no-brainer.

I still somehow feel, that the performance of fast scrolling could be improved by preloading a little bit, instead of waiting until the viewport has actually reached the bottom, but this is more of a tweak than a bugfix and has nothing to do with the original issue (so either use it as feedback, or ignore it for the moment).
